src/Bicep.RegistryModuleTool/ModuleValidators/TestValidator.cs

```diff
@@ -59,7 +59,7 @@ public void Validate(MainBicepTestFile file)
 
             using var tempFileStream = fileSystem.FileStream.CreateDeleteOnCloseStream(tempFilePath);
             var testTemplateElement = JsonElementFactory.CreateElement(tempFileStream);
-            var testDeployments = testTemplateElement.Select($@"$.resources[?(@.type == ""Microsoft.Resources/deployments"" && @.properties.template.metadata._generator.templateHash == ""{this.latestMainArmTemplateFile.TemplateHash}"")]");
+            var testDeployments = testTemplateElement.Select($@"$..resources[?(@.type == ""Microsoft.Resources/deployments"" && @.properties.template.metadata._generator.templateHash == ""{this.latestMainArmTemplateFile.TemplateHash}"")]");
```

Comment:
Was the extra period intentional?

Reply:
Yes, this is to search for the nested deployment recursively.